odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -734,6 +734,54 @@ def test_properties_field_integer_float_boolean(self):
             self.message_1.attributes[2]['value'], False,
             msg='Boolean value must have been converted to False')
 
+        # When the user sets the value 0 for the property fields of type integer
+        # and float, the system should store the value 0 and shouldn't transform
+        # 0 to False (-> unset value).
+
+        self.message_1.attributes = {'int_value': 0, 'float_value': 0}
+        self.assertEqual(len(self.message_1.attributes), 3)
+        self.assertEqual(self.message_1.attributes[0]['value'], 0)
+        self.assertEqual(self.message_1.attributes[1]['value'], 0)
+        self.assertEqual(self.message_1.attributes[2]['value'], False)
+        self.assertEqual(type(self.message_1.attributes[0]['value']), type(0))
+        self.assertEqual(type(self.message_1.attributes[1]['value']), type(0))
+        self.assertEqual(type(self.message_1.attributes[2]['value']), type(False))
+        self.assertEqual(self._get_sql_properties(self.message_1), {'int_value': 0, 'float_value': 0, 'boolean_value': False})
+
+    def test_properties_field_integer_float_falsy_value_edge_cases(self):
+        self.discussion_1.attributes_definition = [
+            {
+                'name': 'int_value',
+                'string': 'Int Value',
+                'type': 'integer',
+                'default': 42
+            }, {
+                'name': 'float_value',
+                'string': 'Float Value',
+                'type': 'float',
+                'default': 0.42
+            }
+        ]
+        message_1 = self.env['test_new_api.message'].create({
+            'discussion': self.discussion_1.id,
+            'author': self.user.id,
+            'attributes': {
+                'int_value': 0,
+                'float_value': 0
```

Comment:
picky std :p (same above)
```suggestion
                'float_value': 0,
```

(but in that case it can be inlined to be consistent with code below)